### PR TITLE
feat(invoice_grace_period): Add draft status to credit notes

### PR DESCRIPTION
**Pull request by rsempe**
## Roadmap Task

👉  https://github.com/getlago/lago/issues/99

## Context

When a billing period is over, users don’t want to invoice straight away.

They want a defined number of days to:
- Collect delayed events for usage
- Adjust invoice amounts for items

## Description

The goal of this PR is to add a new field status to credit notes (`draft` or `finalized`).
This is useful when creating a draft invoice, draft credit note can also be created.
Then, when we're finalizing an invoice the goal of the next PR is to also finalize credit notes.

### CODE REVIEW

File: app/graphql/mutations/credit_notes/void.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Void < BaseMutation
 
       def resolve(id:)
         result = ::CreditNotes::VoidService.new(
-          credit_note: context[:current_user].credit_notes.find_by(id: id),
+          credit_note: context[:current_user].credit_notes.finalized.find_by(id: id),
```
Comment (jdenquin) [inline app/graphql/mutations/credit_notes/void.rb:17]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

**Review by vincent-pochet — APPROVED** [review]
Looks good! Happy to discuss the few comments ;)

File: app/controllers/api/v1/credit_notes_controller.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module V1
     class CreditNotesController < Api::BaseController
       def create
         service = CreditNotes::CreateService.new(
-          invoice: current_organization.invoices.find_by(id: input_params[:invoice_id]),
+          invoice: current_organization.invoices.finalized.find_by(id: input_params[:invoice_id]),
```
[the remark this answers is not in the document]
Comment (rsempe) [reply app/controllers/api/v1/credit_notes_controller.rb:8]: Good spot 👍

File: app/controllers/api/v1/credit_notes_controller.rb [hunk 3]
```diff
@@ -37,12 +37,10 @@ def show
       end
 
       def update
-        credit_note = current_organization.credit_notes.find_by(id: params[:id])
+        credit_note = current_organization.credit_notes.finalized.find_by(id: params[:id])
```
[the remark this answers is not in the document]
Comment (rsempe) [reply app/controllers/api/v1/credit_notes_controller.rb:40]: I agree 👍

File: app/graphql/resolvers/invoice_credit_notes_resolver.rb [hunk 4]
```diff
@@ -16,8 +16,9 @@ class InvoiceCreditNotesResolver < Resolvers::BaseResolver
     def resolve(invoice_id: nil, page: nil, limit: nil)
       validate_organization!
 
-      Invoice.find(invoice_id)
+      Invoice.finalized.find(invoice_id)
```
[the remark this answers is not in the document]
Comment (rsempe) [reply app/graphql/resolvers/invoice_credit_notes_resolver.rb:19]: Indeed, maybe safer to not add this restriction 👍

**Pull request merged**